src/Microsoft.Diagnostics.Monitoring.EventPipe/Configuration/MetricSourceConfiguration.cs

```diff
@@ -13,70 +13,85 @@
 
 namespace Microsoft.Diagnostics.Monitoring.EventPipe
 {
+    [Flags]
+    public enum MetricType
+    {
+        EventCounter = 0x1,
+        Meter = 0x2
+    }
+    
+    public sealed class MetricEventPipeProvider
+    {
+        public string Provider { get; set; }
+
+        public float IntervalSeconds { get; set; }
+
+        public MetricType Type { get; set; } = MetricType.EventCounter | MetricType.Meter;
+    }
+
     public sealed class MetricSourceConfiguration : MonitoringSourceConfiguration
     {
         private readonly IList<EventPipeProvider> _eventPipeProviders;
         public string SessionId { get; private set; }
 
+        public static readonly string[] DefaultProviders = new[] { SystemRuntimeEventSourceName, MicrosoftAspNetCoreHostingEventSourceName, GrpcAspNetCoreServer };
+
         public MetricSourceConfiguration(float metricIntervalSeconds, IEnumerable<string> customProviderNames)
+            : this(metricIntervalSeconds, customProviderNames?.Any() == true ? CreateProviders(metricIntervalSeconds, customProviderNames) :
+            CreateProviders(metricIntervalSeconds, DefaultProviders))
         {
-            RequestRundown = false;
-            if (customProviderNames == null)
-            {
+        }
+
+        public MetricSourceConfiguration(float metricIntervalSeconds, IEnumerable<MetricEventPipeProvider> customProviderNames, int maxHistograms = 20, int maxTimeSeries = 1000) {
```

Comment:
nit: `customProviderNames` might be misleading here (given that it's no longer just names) - maybe it should be `customProviders` or similar?